pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -124,23 +125,20 @@ func (a *AmbientIndex) updateWaypoint(sa model.WaypointScope, ipStr string, isDe
 			}
 			wl.WaypointAddresses = addrs
 			if filtered {
-				// If there was a change, also update the VIPs and record for a push
+				// If there was a change, record for a push
 				updates[model.ConfigKey{Kind: kind.Address, Name: wl.ResourceName()}] = struct{}{}
-				a.byPod[wl.ResourceName()] = wl
-				for vip := range wl.VirtualIps {
-					a.insertWorkloadToService(vip, wl)
-				}
```

Comment:
For this i think we donot need to insert again, it is done in handlePod cb

Reply:
How does this work? We get a Pod update for the waypoint which calls handlePod, but we needto update the user pods

Reply:
In service handler, wl has been updated,

Reply:
But this is for waypoint pod updating, there is no Service involved.

Reply:
`a.insertWorkloadToService(vip, wl)` is actually insert user pod to `byService` right?

When a waypoint updates, it has no influence on it

Reply:
ping @howardjohn